docs/learn/encyclopedia/contract-development/stellar-contract.mdx

```diff
@@ -0,0 +1,142 @@
+---
+sidebar_position: 10
+title: Introduction
+description: Introductiot Smart Contracts
+---
+
+<head>
+  <title>
+    {`The interface that defines objects, functions, and data available to smart
+    contracts.`}
+  </title>
+  <meta charSet="utf-8" />
+  <meta
+    property="og:title"
+    content="The interface that defines objects, functions, and data available to smart contracts."
+  />
+  <meta
+    property="og:description"
+    content="Learn about the interface environment that defines the objects, functions, data sources, and more that are available in smart contracts."
+  />
+</head>
+
+This section provides an overview of Stellar contracts, outlining their key concepts of smart contracts on the Stellar network.
+
+## Smart Contract in WebAssembly (Wasm)
+
+A smart contract on Stellar is a programmable set of executable binary in WebAssembly (Wasm). WebAssembly (Wasm) is a lightweight, portable binary instruction format designed for high-performance execution across various environments, including blockchains, web browsers, and cloud services. In Stellar, Wasm serves as the foundation for smart contracts by enabling a secure and efficient execution environment.
+
+- **Compilation**: Once smart contract is written in a supported language, such as Rust, the contract is compiled into a Wasm executable file. This file contains low-level bytecode that can be efficiently executed within a sandboxed environment.
+- **Upload and stored**: After compilation, the Wasm executable is first uploaded on the Stellar network, where it is stored in a `CONTRACT_DATA` ledger entry. The `CONTRACT_DATA` ledger entry is created to store the Wasm binary data.
+- **Unique hash is created**: An unique identifier for this `CONTRACT_DATA` entry is the hash of the executable file, called `Wasm hash`. To note, this binary executable is stored independently from its deployed contract(s).
+- **Multiple contract instances**: Multiple contract instances can be deployed that references the same Wasm bytecode, with each instances maintaining its own contract storage, state, and configurations.
+
+## Contract Instances
+
+After the executable bytecode is uploaded on-chain, contract instances can be deployed that reference the same bytecode. **A contract Wasm executable can have a one-to-many relationship with its contract instances which function independently.** This means the same executable code can be used by multiple contract instances that all behave identically, because of the shared executable code, while maintaining separate and distinct state data, because the data is tied to the contract instance. A contract instance is stored as its own ledger entry, and any of the contract's [instance storage](#instance-storage) is stored in that same ledger entry alongside the contract instance.
```

Comment:
#instance-storage -> #contract-storage?

Reply:
This is contract instance storage. Not contract storage. It's important to not conflate these.

Reply:
i think jeesun just meant that the `[instance storage](#instance-storage)` link pointed to an anchor on the page (`#instance-storage`) that doesn't exist. you've got a `#contract-storage` section which could be linked to, if you wanted.

Reply:
oh i see you removed the link. thanks!